### PR TITLE
RDPMBT-708 - Upgrade React

### PR DESCRIPTION
**Pull request by heldergoncalves92**

### CODE REVIEW

File: ReactViewResources/package-lock.json [hunk 1]
```diff
@@ -1,146 +1,39 @@
...
-        "object-assign": "^4.1.1"
-      }
-    },
-    "node_modules/react-is": {
-      "version": "16.9.0",
-      "resolved": "https://registry.npmjs.org/react-is/-/react-is-16.9.0.tgz",
-      "integrity": "sha512-tJBzzzIgnnRfEm046qRcURvwQnZVXmuCbscxUO5RWrGTXpon2d4c8mI0D8WE6ydVIm29JiLB6+RslkIvym9Rjw=="
-    }
-  },
+  "lockfileVersion": 1,
```
Comment (joaompneves) [inline ReactViewResources/package-lock.json:3]: upgrade your node/npm

File: Directory.Build.props [hunk 2]
```diff
@@ -2,7 +2,7 @@
   <PropertyGroup>
     <AssemblyVersion>2.0.0.0</AssemblyVersion>
     <FileVersion>2.0.0.0</FileVersion>
-    <Version>2.91.5</Version>
+    <Version>2.92.0</Version>
```
Comment (joaompneves) [inline Directory.Build.props:5]: 91 -> is the version of the webview used .. so dont raise that number

**Review by joaompneves — CHANGES_REQUESTED** [review]

**Review by joaompneves — CHANGES_REQUESTED** [review]

File: Sample.Avalonia/MainView/MainView.tsx [hunk 3]
```diff
@@ -126,6 +127,7 @@ export default class MainView extends React.Component<IMainViewProperties, MainV
                 <button className="tasks-toggle-show" onClick={this.toggleShowTasks}>Show/Block/Hide Tasks</button>
                 {this.renderListView()}
                 <div>{this.state.tasksCount} task(s)</div>
+                <ComponentTest />
```
[the remark this answers is not in the document]
Comment (heldergoncalves92) [reply Sample.Avalonia/MainView/MainView.tsx:130]: At the moment, I'm only testing the build and the runtime.
After everything is fine, I'll revert this.

Currently there are some issues on loading the `react/jsx-runtime`

File: Directory.Build.props (hunk 2 above)
Comment (joaompneves) [reply Directory.Build.props:5]: nope

File: ViewGenerator/tools/package.json [hunk 4]
```diff
@@ -17,7 +17,8 @@
         "@types/webpack": "^4.32.1",
         "@types/webpack-dev-server": "^3.10.1",
         "@types/webpack-manifest-plugin": "^2.0.0",
-        "fork-ts-checker-webpack-plugin": "^6.1.0"
+        "fork-ts-checker-webpack-plugin": "^6.1.0",
+        "typescript": "^4.2.2"
```
Comment (joaompneves) [inline ViewGenerator/tools/package.json:21]: new dep?

Comment (mpm-os) [reply ViewGenerator/tools/package.json:21]: i believe this is not needed here..

**Pull request closed without merging**